### ISSUE TITLE
Mautic 2.14.0 fails to rebuild campaign

### ISSUE THREAD

**Issue by KalleVuorjoki** [mautic/mautic#6460]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | X
| Feature request? | 
| Enhancement? | 

## Description:
Mautic fails to rebuild campaign `mautic:campaigns:rebuild`

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.14.0 and 2.14.1.
| PHP version | PHP 7.1.17-1+ubuntu16.04.1+deb.sury.org+1

PHP has 512M (server has 2G), tried to increase php memory to 1G but process got killed.

### Steps to reproduce:
1. `sudo -u www-data php /var/www/html/mautic/app/console mautic:campaigns:rebuild --campaign-id=1 --batch-limit=5 --max-contacts=10`

 
### Log errors: 

```
11746 total contact(s) to be added in batches of 5
     0/11746 [>---------------------------]   0%PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/html/mautic/app/bundles/LeadBundle/Entity/CustomFieldRepositoryTrait.php on line 92
PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/html/mautic/vendor/symfony/debug/ErrorHandler.php on line 650
```

**Comment by luizeof** [comment]
Please, verify with your host provider to increase your php memory

If you manage your own VPS, this could help you:
https://secure.php.net/manual/pt_BR/ini.core.php#ini.memory-limit

**Comment by luizeof** [comment]
Mautic 2.14.0 has a bug that make an infinite loop on some campaigns ..

can you teste this PR #6364 please?

**Comment by KalleVuorjoki** [comment]
Already tested increasing php memory to 1G without success. Also using `--batch-limit=5 --max-contacts=10` should not require that much memory.

Tested PR #6463, but it didn't help either.

**Comment by StoppingBuck** [comment]
To add to this (I work on the same project as @KalleVuorjoki): Some simple campaigns work fine, and testing has led us to believe that the error only seems to occur with campaigns that involve subtracting one segment from another (see image) -- regardless of which segments we're talking about. Even setting up a campaign with a simple test segment of 3-4 people and asking to exclude an even smaller segment of 1 of said people will result in nothing being sent out at all.

![screen shot 2018-08-22 at 09 27 35](https://user-images.githubusercontent.com/8426548/44454120-f542c180-a5fa-11e8-9b4e-36ff97c47dd4.png)

**Comment by KalleVuorjoki** [comment]
@raymondkent I guess you are missing `>` from cron, so end should be `> /dev/null 2>&1`

**Comment by cswainhart** [comment]
We have the same problem. When we increased memory to 1Gb and added --batch-limit=5 it helped with the small campaigns, but yesterday we had one fail that had 50,000 records and we have no way of getting it going. So we had to send the email out as a segment email instead of a campaign. Hope this gets fixed.

**Comment by raymondkent** [comment]
@KalleVuorjoki  Thank You!  This was so hard to figure out for me, yet such a simple oversight...

**Comment by socmov** [comment]
@heathdutton is there any workaround for now? I tried to manually apply the PR and I still get the same problem.

**Comment by heathdutton** [comment]
@socmov 2.14.1-beta has been released which includes #6463 and #6482. Please give that a try.

If the issue persists, then we'll need more details. Please run the command with `--env=dev` and grab the logs to help pinpoint exactly where you are running out of ram. It feels like there is an accidental recursion happening here, but it's not something I've been able to replicate.

**Comment by socmov** [comment]
Thanks for the update Heath. I'll have a look over the weekend. I actually reverted my instance to 2.13.1 as I needed to send out an email campaign but I'll try upgrade a separate instance over the weekend to test.

**Comment by YosuCadilla** [comment]
I tested with 2.14.1-beta and it seems NOT to solve the issue, here's a complete list of possibly related issues: #6515 #5546 #6482 #6419 #6211 #6433 #6460 #63

Seems Github has issues to solve too: #63 as seen here: mautic/docker-mautic#63

**Comment by YosuCadilla** [comment]
Possibly related too #6321 #6368 #63

**Comment by escopecz** [comment]
Does it happen with all you campaigns or just some? We have to come up with steps to test so devs could configure your problematic scenario, find and fix the problem. Please provide steps to test that contain steps to build the problematic campaign.

**Comment by npracht** [comment]
i think we can consider the fixes #6482 #6211 & #6531 covers this issue. We'll reopen if necessary.
